### ISSUE TITLE
False positive with splat and explicit kwarg

### ISSUE THREAD

**Issue by torfsen** [python/mypy#20441]
**Bug Report**

Mypy reports multiple values for a kwarg if args are given via splat and the kwarg is explicitly set.

**To Reproduce**

```python
def foo(x: int, y: int, z: str) -> None:
    print(x, y, z)

args = [1, 2]
foo(*args, z="...")
```

**Expected Behavior**

Mypy should not report any problems.

**Actual Behavior**

```
mypysandbox.py:5: error: "foo" gets multiple values for keyword argument "z"  [misc]
mypysandbox.py:5: error: Argument 1 to "foo" has incompatible type "*list[int]"; expected "str"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

```
$ mypy --version
mypy 1.19.1 (compiled: yes)
```

```
$ python --version
Python 3.12.12
```

Mypy config from `pyproject.toml`:
```toml
[tool.mypy]
strict = true
allow_redefinition =  true
```

**Comment by torfsen** [comment]
This is very similar to #10481, but here it's splat + kwarg instead of splat + splat. I wasn't sure if this difference is relevant, feel free to close as duplicate.

There is also #2582, but that is very old and has long been fixed.

#10348 looks like it's the same issue as this one and #10481 but reported just for `datetime.datetime`.

And then there is #6799, which is also very similar.

So definitely some duplicates, but I don't know enough about the details to figure out which one should be the main ticket.

**Comment by A5rocks** [comment]
FWIW I'm tempted to say this isn't a bug, because the list could technically be 3 items long and mypy wouldn't know any better. Let's see what others think...

**Comment by randolf-scholz** [comment]
In the provided example, using a tuple instead of a list fixes the issue. The question is what to do in a case like

```pyhton
def foo(x: int, y: int, z: str) -> None: ...

def test(items: list[int]) -> None:
    foo(*items, z="string")
```

FWIW `ty` and `pyrefly` also error here, while `pyright` doesn't There is a trade-off:

1. First match the keyword argument, and match the positionals afterwards. This avoids the false positive if the list happens to be the correct length, but produces a false negative if the list has incorrect length
2. First match the positional, and then the keyword args (this is mypy currently does), which produces a false negative when the list happens to have the correct length.

**Comment by sterliakov** [comment]
I'm +1 on keeping this as-is - passing arguments like this is a relatively common source of bugs, just as bad as `func(**kwargs, foo=bar)`. Fixed-size tuples should be allowed (they provide enough information), but not an arbitrary iterable. 

But current mypy behavior is inconsistent - we do not report too few arguments for `func(*args)`, for example. We need to agree where the line is drawn and align all such too many/too few args checks for `*args` and `**kwargs` behavior.
